File: 1929-concatenation-of-array/1929-concatenation-of-array.c


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* getConcatenation(int* nums, int numsSize, int* returnSize){
   
    
    int numsSize2 = numsSize * 2;
    
    int* arr =(int*)malloc((numsSize2)*sizeof(int)); 
    
    for (int i = 0;i < numsSize ; i++){
        arr[i] = nums[i];
        arr[i+numsSize] = nums[i];
    }
    *returnSize = numsSize2;
    
    return arr;
}

File: 1929-concatenation-of-array/1929-concatenation-of-array.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int numsLength = nums.length;
        int[] arr = new int[numsLength*2];
        for (int i = 0; i < arr.length;i++){
            if (i <numsLength){
                arr[i] = nums[i];
            }else{
                arr[i] = nums[i-numsLength];
            }
        }
        return arr;
    }
}